docs/src/main/sphinx/security/opa-access-control.md

```diff
@@ -212,6 +218,99 @@ The `targetResource` is used in cases where a new resource, distinct from the on
   }
 }
 ```
+## RowFilters 
+Row filtering allows Trino to filter out some rows from the result before
+returning it to the caller, controlling what data different users can see. This
+plugin supports retrieving filters from OPA.
+
+For these policies to be enabled, make sure to set `opa.policy.row-filters-uri`
+in the OPA plugin configuration.
+
+For instance, an OPA policy for row filtering may be implemented as below:
+
+```rego
+  package trino
+  import future.keywords.in
+  import future.keywords.if
+  import future.keywords.contains
+
+  default allow := true
+
+  table_resource := input.action.resource.table
+  is_admin {
+    input.context.identity.user == "admin"
+  }
+
+  rowFilters contains {"expression": "user_type <> 'customer'"} if {
+      not is_admin
+      table_resource.catalogName == "sample_catalog"
+      table_resource.schemaName == "sample_schema"
+      table_resource.tableName == "restricted_table"
+  }
+```
+The response expected by the plugin is an array of objects, each of them in the
+format `{"expression":"clause"}`.
+
+Each expression essentially behaves like an additional `WHERE` clause.
+
+
+Also, It is possible for **multiple** row filters to be returned for a single
+OPA request, they will all be applied.
+
+About "identity"
+- each object may contain an identity field. The identity field allows Trino to
+  evaluate these row filters under a **different** identity - such that a filter
+  can target a column the requesting user cannot see.
+ 
+
+## Column masking
+
+Column masking allows Trino to mask out/obscure the data in some columns for
+specific users, without outright denying access. This plugin supports fetching
+column masks from OPA.
+
+For these policies to be enabled, make sure to set
+`opa.policy.column-masking-uri` in the opa-plugin configuration.
+
+For instance, a policy configuring column masking may be implemented as below:
+
+```rego
+  package trino
+  import future.keywords.in
+  import future.keywords.if
+  import future.keywords.contains
+
+  default allow := true
+
+  column_resource := input.action.resource.column
+  is_admin {
+    input.context.identity.user == "admin"
+  }
+
+  columnMask := {"expression": "NULL"} if {
+      not is_admin
+      column_resource.catalogName == "sample_catalog"
+      column_resource.schemaName == "sample_schema"
+      column_resource.tableName == "restricted_table"
+      column_resource.columnName == "user_phone"
+  }
+
+  columnMask := {"expression": "'****' || substring(user_name, -3)"} if {
+      not is_admin
+      column_resource.catalogName == "sample_catalog"
+      column_resource.schemaName == "sample_schema"
+      column_resource.tableName == "restricted_table"
+      column_resource.columnName == "user_name"
+  }
+```
+
+Unlike row filtering, only a **single column mask** may be returned for a given
```

Comment:
Same as above .. explain what the script does some more and refactor the section here and following into normal sentences.